c7n/actions/policy.py

```diff
@@ -41,6 +42,11 @@ def remove_statements(match_ids, statements, matched=()):
     return statements, found
 
 
+def statement_id(s):
+    # for statements without a sid, use a checksum for identity
+    return hashlib.sha224(utils.dumps(s, indent=0).encode('utf8')).hexdigest()
```

Comment:
Do we care if this always returns the same value for a given statement? If so, we might want to include `sort_keys=True` so key reordering doesn't change the hash.

...if we don't care because we just want _something_ here, then this seems fine as is :+1: .

Reply:
That seems right. Ok non-issue then :)